File: 1-2.rb
#!/usr/bin/env ruby

readings = File.read('1.input').lines.map(&:to_i)

increments = 0
win_size = 3
last = readings[0..win_size - 1].sum

1.upto(readings.size - win_size) do |i|
  win_sum = readings[i..i + win_size - 1].sum
  increments += 1 if win_sum > last
  last = win_sum
end

print increments, "\n"

File: 1-2.py
#!/usr/bin/env python

readings = [int(row) for row in open('1.input').readlines()]

increments = 0
win_size = 3
last = sum(readings[0:win_size])

for i in range(1, len(readings)):
    win_sum = sum(readings[i:i + win_size])
    if win_sum > last:
        increments += 1
    last = win_sum

print(increments)
